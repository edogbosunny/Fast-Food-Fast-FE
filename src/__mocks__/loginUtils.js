/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  user: {},
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  loginUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockedLoginState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};

export const nextProps = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
};

export const Loginstate = {
  email: '',
  password: '',
};
