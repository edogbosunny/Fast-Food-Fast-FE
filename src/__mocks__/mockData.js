const mock = {
  loginDetails: {
    email: 'test@test.com',
    password: '123456',
  },
  registerDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',
    lastname: 'test,',
  },
  errRegisterDetails: {
    email: 'test1@test.com',
    password: 123456,
    confirmpassword: 123456,
    firstname: 'test',

  },
};
export default mock;
