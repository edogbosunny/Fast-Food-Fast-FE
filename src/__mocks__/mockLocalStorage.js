let localStorage = {};

export default {
  setItem(key, value) {
    return Object.assign(localStorage, { [key]: value });
  },
  removeItem(key) {
    return delete localStorage[key];
  },
  getItem(key) {
    return localStorage[key];
  },
  clear() {
    localStorage = {};
  },
};
