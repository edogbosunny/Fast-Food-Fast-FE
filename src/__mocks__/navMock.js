/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
  history: {
    action: 'PUSH',
    push: jest.fn(),
  },
  location: {
    hash: '',
    pathname: '/login',
    search: '',
  },
  logoutUser: jest.fn(),
  match: { path: '/login', url: '/login', isExact: true },
};

export const mockState = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: { id: 1, iat: 1549439277, exp: 1549525677 },
  },
};
