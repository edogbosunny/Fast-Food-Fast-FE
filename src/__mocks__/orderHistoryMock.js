/* eslint-disable import/prefer-default-export */
export const props = {
  dispatch: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      mealItem: [{}],
      message: 'Food order retrieved succesfully',
    },
    getAllOrders: jest.fn(),
    getUserOrderById: jest.fn(),
  },
};

export const tableProps = {
  getAllOrders: jest.fn(),
  getUserOrderById: jest.fn(),
  orderHistory: {
    loading: false,
    orderHistory: {
      count: 30,
      data: [{}],
      message: 'Food order retrieved succesfully',
    },
  },
};
