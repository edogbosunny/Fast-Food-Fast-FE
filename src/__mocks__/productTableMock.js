/* eslint-disable import/prefer-default-export */
export const props = {
  cart: {
    count: 800,
    cart: {
      0: {
        meal_id: 1, price: '600', meal: 'Spaniish Noodles', qty: 1,
      },
    },
    items: null,
    cartCount: 800,
  },
  checkoutOrder: jest.fn(),
  order: {
    order: {},
  },
  history: {
    push: jest.fn(),
  },
  removeFromCart: jest.fn(),
};
