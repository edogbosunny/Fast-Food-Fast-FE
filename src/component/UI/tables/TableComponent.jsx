/* eslint-disable class-methods-use-this */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import { connect } from 'react-redux';

class TableComponent extends Component {
  constructor() {
    super();
    this.state = {
      cart: {},
    };
    this.onDelete = this.onDelete.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      cart: nextProps.cart,
    });
  }

  onDelete(e) {
    console.log(e.target.id);
    // const { cart } = this.props.cart;
    // Object.values(cart).map((orderData) => {
    //   console.log(orderData);
    //   if (e.target.id) {
    //     orderData.delete(orderData);
    //   }
    // });
  }


  render() {
    const { cart } = this.props.cart;
    console.log('state---->', cart);
    // return(<div>jj</div>)
    return Object.values(cart).map((orderData) => {
      // const menu = allCartItems[menuId];
      console.log(orderData.meal);
      return (
        <tr key={orderData.meal_id}>
          <td>
            {orderData.meal}
            {' '}
            <span className='m-3'>{orderData.qty}</span>
            <span>
              <button onClick={this.onDelete} id={orderData.meal_id} type='button' className='fas fa-times del-cart' />
            </span>
          </td>
          <td className='float-right'>
            {orderData.price}
          </td>
        </tr>
      );
    });
  }
}

const mapStateToProps = state => ({
  cart: state.cart,
});
export default connect(mapStateToProps)(TableComponent);
