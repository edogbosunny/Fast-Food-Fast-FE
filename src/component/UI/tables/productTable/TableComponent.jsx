/* eslint-disable arrow-body-style */
/* eslint-disable no-shadow */
/* eslint-disable class-methods-use-this */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
/* eslint-disable react/no-unused-state */
import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { removeFromCart } from '../../../../actions/cartActions';

export class TableComponent extends Component {
  constructor() {
    super();
    this.state = {
      cart: {},
    };
    this.onDelete = this.onDelete.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      cart: nextProps.cart,
    });
  }

  onDelete(e) {
    // console.log(e.target.id);
    const { cart } = this.props.cart;
    const { removeFromCart } = this.props;
    const thisCart = cart;
    const key = Object.keys(thisCart).filter(newKey => newKey === e.target.id);
    const newCart = thisCart;
    // console.log('OLDCart', cart);
    delete newCart[key[0]];
    // console.log('NewCart', newCart);
    // console.log('state', this.state);
    removeFromCart(newCart);
  }

  render() {
    const { cart } = this.props.cart;
    // console.log('state---->', cart);
    // return(<div>jj</div>)
    return Object.values(cart).map((orderData) => {
      // const menu = allCartItems[menuId];
      // console.log(orderData.meal);
      return (
        <Fragment key={orderData.meal_id}>
          <tr>
            <td data-label='Order'>
              {orderData.meal}
              {' '}
              <span>
                <button onClick={this.onDelete} id={orderData.meal_id} type='button' className='fas fa-times del-cart' />
              </span>

            </td>
            <td data-label='Cost'>
              {' '}
              {orderData.qty}
            </td>
          </tr>
        </Fragment>
      );
    });
  }
}

export const mapStateToProps = state => ({
  cart: state.cart,
  loading: state.loader,
});
export default connect(mapStateToProps, { removeFromCart })(TableComponent);
