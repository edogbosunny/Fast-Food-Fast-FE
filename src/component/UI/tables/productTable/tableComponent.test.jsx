import React from 'react';
import { shallow } from 'enzyme';
import { props } from '../../../../__mocks__/productTableMock';
import { TableComponent, mapStateToProps } from './TableComponent';

describe('orderHistore component test-suit', () => {
  const wrapper = shallow(<TableComponent {...props} />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should test mapStateToProps', () => {
    const state = mapStateToProps(props);
    expect(state).toBeTruthy();
  });

  // sence
  it('should test onDelete', () => {
    const event = {
      target: { name: 'id', value: {} },
      preventDefault: () => { },
    };
    wrapper.instance().onDelete(event);
    expect(wrapper.instance().state.cart).toEqual(event.target.value);
  });

  it('should test componentwillrecieveprops()', () => {
    const componentWillReceivePropsSpy = jest.spyOn(TableComponent.prototype, 'componentWillReceiveProps');
    const shallowWrapper = shallow(<TableComponent {...props} />);
    shallowWrapper.instance().componentWillReceiveProps(props);
    expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
  });
});
