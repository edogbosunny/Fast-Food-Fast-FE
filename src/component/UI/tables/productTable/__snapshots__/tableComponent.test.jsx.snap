// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderHistore component test-suit should match snapshot 1`] = `
<Fragment
  key="1"
>
  <tr>
    <td
      data-label="Order"
    >
      Spaniish Noodles
       
      <span>
        <button
          className="fas fa-times del-cart"
          id={1}
          onClick={[Function]}
          type="button"
        />
      </span>
    </td>
    <td
      data-label="Cost"
    >
       
      1
    </td>
  </tr>
</Fragment>
`;
